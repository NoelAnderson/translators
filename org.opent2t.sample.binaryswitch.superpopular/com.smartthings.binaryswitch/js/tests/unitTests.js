var test = require('ava');
var translatorPath = require('path').join(__dirname, '..');
var runBinarySwitchTests = require('opent2t-device-binaryswitch/binarySwitchTests');
var deviceData = require('./devicedata');
var OpenT2TLogger = require('opent2t').Logger;
var logger = new OpenT2TLogger("info");
var MockHub = require('opent2t-device-smartthingshub/mockSmartthingsHub');
var mockHub = new MockHub(logger, deviceData);

var settings = {
    logger,
    createTranslator: mockHub.createTranslator(translatorPath, deviceData.base_state.id),
    test: test,
    setTestData: mockHub.setTestData
};

// Run standard binary switch unit tests
runBinarySwitchTests(settings);