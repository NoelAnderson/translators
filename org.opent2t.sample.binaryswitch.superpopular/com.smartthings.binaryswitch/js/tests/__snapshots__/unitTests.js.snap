exports[`GetPlatform 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.binaryswitch.superpopular\",\"translator\":\"opent2t-translator-com-smartthings-binaryswitch\",\"controlId\":\"3b5e69d7-1e74-4bce-8b5d-da9ead378a56\",\"endpointUri\":\"https://sample.endpoint.com/123\"},\"availability\":\"online\",\"pi\":\"3b5e69d7-1e74-4bce-8b5d-da9ead378a56\",\"mnmn\":\"Mock Manufacturer\",\"mnmo\":\"Mock Model\",\"n\":\"Mock Name\",\"rt\":[\"org.opent2t.sample.binaryswitch.superpopular\"],\"entities\":[{\"n\":\"Mock Name\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"rt\":[\"oic.d.smartplug\"],\"di\":\"80dc960f-157a-47a1-385b-b3642a26ec2b\",\"resources\":[{\"href\":\"/power\",\"rt\":[\"oic.r.switch.binary\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.binaryswitch.superpopular","translator":"opent2t-translator-com-smartthings-binaryswitch","controlId":"3b5e69d7-1e74-4bce-8b5d-da9ead378a56","endpointUri":"https://sample.endpoint.com/123"},"availability":"online","pi":"3b5e69d7-1e74-4bce-8b5d-da9ead378a56","mnmn":"Mock Manufacturer","mnmo":"Mock Model","n":"Mock Name","rt":["org.opent2t.sample.binaryswitch.superpopular"],"entities":[{"n":"Mock Name","icv":"core.1.1.0","dmv":"res.1.1.0","rt":["oic.d.smartplug"],"di":"80dc960f-157a-47a1-385b-b3642a26ec2b","resources":[{"href":"/power","rt":["oic.r.switch.binary"],"if":["oic.if.a","oic.if.baseline"],"id":"power","value":true}]}]}"`;
